content/en/platform/corda/5.0-dev-preview-2/getting-started/first-flow/code-java.md

```diff
@@ -0,0 +1,458 @@
+* `MyFirstFlowStartArgs` — provides a wrapper around the single arguments that need to be passed into the flow — the other member of the application network who the message should be sent to:
+   ```java
+   public class MyFirstFlowStartArgs {
+    public MemberX500Name otherMember;
+
+    public MyFirstFlowStartArgs(MemberX500Name otherMember) {
+        this.otherMember = otherMember;
+    }
+
+    // The JSON Marshalling Service, which handles serialisation, needs this constructor.
+    public MyFirstFlowStartArgs() {}
+   }
+   ```
+   The [MyFirstFlowTest](../fast-feedback-with-the-simulator/fast-feedback-with-the-simulator.html) Simulator example uses this class.
+* `Message` —  specifies the sender and the message. This is used for both the message sent from the initiator to the responder and subsequently the message sent back from the responder to the initiator. Note, as this is a class defined in a CorDapp and it is going to be sent ‘down the wire’ between two virtual nodes, it requires the `@CordaSerializable` annotation.
+   ```java
+   @CordaSerializable
+   public class Message {
+       public Message(MemberX500Name sender, String message) {
+           this.sender = sender;
+           this.message = message;
+       }
+
+       public MemberX500Name getSender() {
+           return sender;
+       }
+
+       public String getMessage() {
+           return message;
+       }
+
+       public MemberX500Name sender;
+       public String message;
+   }
+   ```
+## Initiating and Responding Flows
+To trigger a flow from HTTP-RPC, the flow must  inherit from `RPCStartableFlow`. Most flows will come in pairs; one initiating flow and a corresponding responder flow. The responder flow must inherit from `ResponderFlow`. The two flows are linked by adding the `@InitiatingFlow` and `@InitiatedBy` annotations which both specify the same protocol in this case "my-first-flow":
+```java
+@InitiatingFlow(protocol = "my-first-flow")
+public class MyFirstFlow implements RPCStartableFlow  { ... }
+```
+```java
+@InitiatedBy(protocol = "my-first-flow")
+public class MyFirstFlowResponder implements ResponderFlow  { ... }
+```
+
+## Logging
+
+It is useful to add logging statements to both the initiating and responder flows.
+To do this, add a logger to each class.
+When running tests locally, the console displays log entries.
+When running on Corda, the log files are updated.
+```java
+    private final Logger log = LoggerFactory.getLogger(MyFirstFlowResponder.class);
+```
+The log files for CSDE are located in the logs folder in the root of the project.
+The CSDE starts a new log file each time a new instance of Corda is created or when the log file grows too large.
+Because the Corda combined worker runs all of the Corda processes in one JVM process, there are a lot of log entries.
+We recommend adding an easily searchable tag to each log message. For example:
+```kotlin
+        log.info("MFF: MyFirstFlow.call() called")
```

Comment:
Java has a semicolon at the end of the line:
`        log.info("MFF: MyFirstFlow.call() called");`